pkg/query/arrow.go

```diff
@@ -138,3 +138,11 @@ func (a *Arrow) Explain() Explain {
 		SubExplain: []Explain{a.left.Explain(), a.right.Explain()},
 	}
 }
+
+func (a *Arrow) Subiterators() []Iterator {
+	return []Iterator{a.left, a.right}
+}
+
+func (a *Arrow) ReplaceSubiterators(newSubs []Iterator) Iterator {
+	return &Arrow{left: newSubs[0], right: newSubs[1]}
+}
```

Comment:
Curious about the signature here - this seems like it could be a pure function? Is it more about code organization and ensuring that an interface is satisfied?

Oh. I guess it would be different if golang had overloads.

Reply:
Yep; and the interface will come in handy when we start rewriting trees